sycl/test/type_traits/half_operator_types.cpp

```diff
@@ -0,0 +1,97 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+//==-------------- type_traits.cpp - SYCL type_traits test -----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+
+template <typename T1, typename T_rtn>
+void check_half_math_operator_types(sycl::queue Queue) {
+
+  // Test on host
+  static_assert(std::is_same_v<decltype(T1(1) + sycl::half(1)), T_rtn>);
```

Comment:
nit: that could also be (I think) 

```
decltype(std::declval<T1>() + std::declval<sycl::half>())
```

But it's probably not worth decreased signal/noise ratio to be that pedantic here.

Reply:
Done